src/model.spec.ts

```diff
@@ -0,0 +1,128 @@
+import fc from 'fast-check';
```

Comment:
I think I'd add one overall test that checks the way you would actually intend the developer to use these functions. There's a bunch of test and parse functions and these unit tests are great. What's the steps for using them? An integration test for this module, if you will.

As well as testing the implementation that the program would use, this would serve to document the way to do the JSON parsing / validation.